src/editor/components/EditorElement.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * ---------------------------------------------------------------------
+ * Copyright (c) 2020 EclipseSource Munich
+ * Licensed under MIT
+ * https://github.com/eclipsesource/jsonforms-editor/blob/master/LICENSE
+ * ---------------------------------------------------------------------
+ */
+
+import { Grid, makeStyles } from '@material-ui/core';
+import React from 'react';
+import { useDrag } from 'react-dnd';
+
+import { DndItems } from '../../core/dnd';
+import { getIconForUISchemaType } from '../../core/icons';
+import {
+  getUISchemaPath,
+  LinkedUISchemaElement,
+} from '../../core/model/uischema';
+
+export interface EditorElementProps {
+  wrappedElement: LinkedUISchemaElement;
+}
+
+const useEditorElementStyles = makeStyles((theme) => ({
+  editorElement: (props: { isDragging: boolean }) => ({
+    alignItems: 'center',
+    justifyContent: 'center',
+    border: '1px solid #D3D3D3',
+    padding: theme.spacing(1),
+    opacity: props.isDragging ? 0.5 : 1,
+  }),
+  iconGridItem: {
+    alignSelf: 'flex-start',
+  },
+}));
+
+export const EditorElement: React.FC<EditorElementProps> = ({
+  wrappedElement,
+  children,
+}) => {
+  const [{ isDragging }, drag] = useDrag({
+    item: DndItems.dragUISchemaElement(wrappedElement),
+    collect: (monitor) => ({
+      isDragging: !!monitor.isDragging(),
+    }),
+  });
+  const classes = useEditorElementStyles({ isDragging });
+  return (
+    <Grid
+      item
+      container
+      wrap='nowrap'
+      className={classes.editorElement}
+      ref={drag}
+      xs
+    >
+      <Grid
+        item
+        key={`${getUISchemaPath(wrappedElement)}-${0}-icon`}
+        className={classes.iconGridItem}
+      >
+        {React.createElement(getIconForUISchemaType(wrappedElement.type), {})}
+      </Grid>
```

Comment:
This renders the icon and right to it the control which leads to a lot of whitespace to the left of the control. I would prefer to render two rows, where the icon is in the top row (left) and the control is in the second row grabbing full space.